deltacat/utils/metrics.py

```diff
@@ -38,8 +42,63 @@ def __init__(
         self.region = region
         self.metrics_target = metrics_target
         self.metrics_namespace = metrics_namespace
-        self.metrics_dimensions = metrics_dimensions
-        self.metrics_kwargs = metrics_kwargs
+
+        # Pyrsistent is unable to enforce immutability on nested objects
+        # Please avoid modifying stored objects to preserve read-only status
+        self.metrics_dimensions = pyrsistent.v(*metrics_dimensions)
```

Comment:
This solution is not complete for enforcing immutability.

Reply:
Do you mean for just `metrics_dimensions` or also for `metrics_kwargs`? I suppose for `metrics_dimensions` you could enforce perfect immutability, but `metrics_kwargs` is unenforceable due to its open-endedness.

Reply:
After discussion, we decided to remove pyrsistent because it doesn't actually solve the problem of immutability, it just adds bloat to our total packages.